src/test/java/edu/wisc/my/messages/model/ValidDateTest.java

```diff
@@ -40,4 +41,67 @@ public void nullDatesPass() {
         assertTrue(message.isValidToday());
     }
 
+  /**
+   * Test that messages that expire later today are not considered expired.
+   * That is, that expiration supports the THH:MM suffix on ISO date-times.
+   */
+  @Test
+  public void expiringLaterTodayIsNotExpired() throws InterruptedException {
+
+    // assumption: this test will not take more than 2 seconds.
+
+    // this test relies on two seconds from now being in the same date, so that an implementation
+    // that respects time will notice that the that timestamp is not yet expired whereas an
+    // implementation that only respects dates will consider that timestamp expired.
+    //
+    // therefore edge case: it is less than four seconds before the date rolls over
+    // in this case wait four seconds to escape the edge case
+
+    LocalDateTime now = LocalDateTime.now();
+
+    if (now.getHour() == 23 && now.getSecond() > 56) {
```

Comment:
The Cinderella rule of testing.

Reply:
Heh. Failed to check minutes. This is bugged.